velocity normVelocity
moveConfiguration moveConfiguration
---
bool res
